src/isolation.rs

```diff
@@ -0,0 +1,61 @@
+use std::{collections::HashMap, ffi::OsString, fs, path::PathBuf};
+
+/// HashMap matching a path in the guest OS ([String]) a path in the host OS ([OsString]).
+pub struct UhyveFileMap {
+	files: HashMap<String, OsString>,
+}
+
+impl UhyveFileMap {
+	/// Creates a UhyveFileMap.
+	///
+	/// Using a list of parameters stored in a [Vec<String>], this function creates
+	/// a HashMap that can match a path on the host operating system given a path on
+	/// the guest operating system.
+	///
+	/// See [crate::hypercall::open] to see this in practice.
+	///
+	/// * `parameters` - A list of parameters with the format `./host_path.txt:guest.txt`
+	pub fn new(parameters: &[String]) -> Option<UhyveFileMap> {
+		Some(UhyveFileMap {
+			files: parameters
+				.iter()
+				.map(String::as_str)
+				.map(Self::split_guest_and_host_path)
+				.map(|(guest_path, host_path)| {
+					(
+						guest_path,
+						fs::canonicalize(&host_path).map_or(host_path, PathBuf::into_os_string),
+					)
+				})
+				.collect(),
+		})
+	}
+
+	/// Separates a string of the format "./host_dir/host_path.txt:guest_path.txt"
+	/// into a guest_path (String) and host_path (OsString) respectively.
+	///
+	/// Keep in mind that the order of the parameters is the inverse of the one
+	/// in the actual HashMap itself, as we want to use the guest_path as a key
+	/// to look up the respective host_path, as well as provide an intuitive
+	/// interface reminiscent of other VMMs like Docker's.
+	///
```

Comment:
This is an internal detail, and imho does not need to be part of the functions' documentation.

Reply:
https://github.com/hermit-os/uhyve/commit/e12bdc6fc3e1a4f83cc57aec8e87b6e9b013a91f